livekit-agents/livekit/agents/beta/tools/http/send_http.py

```diff
@@ -0,0 +1,306 @@
+import asyncio
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any
+
+import aiohttp
+
+from .... import FunctionTool
+from ....llm.tool_context import ToolError, function_tool
+from ....log import logger
+from ....voice import AgentSession, RunContext
+
+
+@dataclass
+class HTTPResponse:
+    """Structured response from HTTP request.
+
+    Attributes:
+        success: Whether the request was successful (2xx status)
+        status_code: HTTP status code
+        body: Response body as string
+        headers: Response headers
+        error: Error message (if failed)
+    """
+
+    success: bool
+    status_code: int
+    body: str
+    headers: dict[str, str]
+    error: str | None = None
+
+
+@dataclass
+class HTTPToolConfig:
+    """Configuration for HTTP tool customization.
+
+    Attributes:
+        url: Endpoint URL (required)
+        name: Custom tool name (optional)
+        description: Custom tool description (optional)
+        method: HTTP method (default: "POST")
+        headers: Optional dict of headers to send with request
+        timeout_ms: Request timeout in milliseconds (default: 30000)
+        parameters: JSON schema dict for tool parameters (defaults to empty object schema)
+        execution_message: Optional message to announce before executing (spoken by agent)
+        output_normalizer: Optional function to transform HTTPResponse to string or None
+            - Return string (including ""): agent uses it in next response
+            - Return None: agent stays silent (good for background calls)
+    """
+
+    url: str
+    name: str | None = None
+    description: str | None = None
+    method: str = "POST"
+    headers: dict[str, str] | None = None
+    timeout_ms: int = 30000
+    parameters: dict[str, Any] | None = None
+    execution_message: str | None = None
+    output_normalizer: Callable[[HTTPResponse], str | None] | None = None
+
+
+async def _announce_execution(tool_name: str, message: str, session: AgentSession) -> None:
+    """Announce execution message via agent session.
+
+    Args:
+        tool_name: Name of the tool being executed
+        message: Message to announce
+        session: AgentSession from RunContext
+    """
+    try:
+        await session.generate_reply(
+            instructions=f"You are running {tool_name} tool (do not announce the tool name) for user and should announce it using this instruction: {message}",
+            allow_interruptions=False,
+        )
+    except Exception as e:
+        logger.debug(
+            "failed to announce execution for tool",
+            extra={"tool_name": tool_name, "error": str(e)},
+        )
+
+
+async def _make_http_request(
+    url: str,
+    method: str,
+    headers: dict[str, str] | None,
+    timeout_ms: int,
+    arguments: dict[str, Any],
+) -> HTTPResponse:
+    """Make HTTP request with given parameters.
+
+    Args:
+        url: Endpoint URL
```

Comment:
in agent builder work I found that it's a must-have to support templating arguments into named path segments if this is going to work for any REST API. so this shouldn't be an actual URL but should be a URL template, and we'd need a way to partition the passed arguments into those that go into the URL and those that go into the payload